docs/examples/image_features/glcm.jl

```diff
@@ -0,0 +1,139 @@
+# ---
+# cover: assets/glcm.gif
+# title: Gray Level Co-occurrence Matrix
+# description: This demo shows GLCM(Gray Level Co-occurrence Matrix)
+# author: Ashwani Rathee
+# date: 2021-08-9
+# ---
+
+# Gray Level Co-occurrence Matrix (GLCM) is used for texture analysis. 
+# We consider two pixels at a time, called the reference and the neighbour pixel.
+# We define a particular spatial relationship between the reference and neighbour 
+# pixel before calculating the GLCM. For eg, we may define the neighbour to be 1 
+# pixel to the right of the current pixel, or it can be 3 pixels above, or 2 pixels
+# diagonally (one of NE, NW, SE, SW) from the reference.
+
+# Once a spatial relationship is defined, we create a GLCM of size (Range of
+# Intensities x Range of Intensities) all initialised to 0. For eg, a 8 bit single
+# channel Image will have a 256x256 GLCM. We then traverse through the image and for
+# every pair of intensities we find for the defined spatial relationship, we increment
+# that cell of the matrix.
+
+# ## Gray Level Co-occurence Matrix
+
+# Each entry of the GLCM[i,j] holds the count of the number of times that pair of 
+# intensities appears in the image with the defined spatial relationship.
```

Comment:
Hmmm, I actually still don't quite understand the contents of `GLCM` here. Maybe add more explanation here, for example, why is `GLCM[1, 3] == 3` in this case? (I'm using 1-based indexing but this figure seems using 0-based indexing).

Reply:
Maybe also add this and explain the arguments to `glcm`.

```julia
X = [0 1 0 2;
     0 2 1 1;
     3 1 0 0;
     0 0 2 3]
glcm(X, 1, 0, 4)
```